include/nbl/builtin/hlsl/bxdf/common.hlsl

```diff
@@ -48,358 +174,738 @@ struct Basic
+        retval.BdotH = (tangentSpaceV.y + tangentSpaceL.y) * LplusV_rcpLen;
+
+        return retval;
+    }
+    template<class AnisotropicInteraction, class LS NBL_FUNC_REQUIRES(surface_interactions::Anisotropic<AnisotropicInteraction> && LightSample<LS>)
+    static this_t createForReflection(
+        NBL_CONST_REF_ARG(AnisotropicInteraction) interaction,
+        NBL_CONST_REF_ARG(LS) _sample)
+    {
+        return createForReflection(interaction.getTangentSpaceV(), _sample.getTangentSpaceL(), _sample.getVdotL());
+    }
+    // transmissive cases need to be checked if the path is valid before usage
+    static this_t create(
+        NBL_CONST_REF_ARG(vector3_type) V, NBL_CONST_REF_ARG(vector3_type) L,
+        NBL_CONST_REF_ARG(vector3_type) T, NBL_CONST_REF_ARG(vector3_type) B, NBL_CONST_REF_ARG(vector3_type) N,
+        NBL_CONST_REF_ARG(fresnel::OrientedEtas<scalar_type>) orientedEtas, NBL_REF_ARG(vector3_type) H
+    )
+    {
+        this_t retval;
+        retval.iso_cache = isocache_type::create(V,L,N,orientedEtas,H);
+        const bool valid = retval.iso_cache.NdotH >= 0.0;
+        if (valid)
+        {
+            retval.TdotH = nbl::hlsl::dot<vector3_type>(T,H);
+            retval.BdotH = nbl::hlsl::dot<vector3_type>(B,H);
+        }
+        return retval;
+    }
+    template<class AnisotropicInteraction, class LS NBL_FUNC_REQUIRES(surface_interactions::Anisotropic<AnisotropicInteraction> && LightSample<LS>)
+    static this_t create(
+        NBL_CONST_REF_ARG(AnisotropicInteraction) interaction,
+        NBL_CONST_REF_ARG(LS) _sample,
+        NBL_CONST_REF_ARG(fresnel::OrientedEtas<scalar_type>) orientedEtas
+    )
+    {
+        this_t retval;
+        vector3_type H;
+        retval.iso_cache = isocache_type::template create<typename AnisotropicInteraction::isotropic_interaction_type, LS>(interaction.isotropic,_sample,orientedEtas,H);
+        const bool valid = retval.iso_cache.NdotH >= 0.0;
+        if (valid)
+        {
+            retval.TdotH = nbl::hlsl::dot<vector3_type>(interaction.getT(),H);
+            retval.BdotH = nbl::hlsl::dot<vector3_type>(interaction.getB(),H);
+        }
+        return retval;
+    }
+
+    bool isValidVNDFMicrofacet(const bool is_bsdf, const bool transmission, const scalar_type VdotL, const scalar_type eta, const scalar_type rcp_eta)
+    {
+        return iso_cache.isValidVNDFMicrofacet(is_bsdf, transmission, VdotL, eta, rcp_eta);
+    }
+
+    scalar_type getVdotH() NBL_CONST_MEMBER_FUNC { return iso_cache.VdotH; }
+    scalar_type getLdotH() NBL_CONST_MEMBER_FUNC { return iso_cache.LdotH; }
+    scalar_type getNdotH() NBL_CONST_MEMBER_FUNC { return iso_cache.NdotH; }
+    scalar_type getNdotH2() NBL_CONST_MEMBER_FUNC { return iso_cache.NdotH2; }
+
+    scalar_type getTdotH() NBL_CONST_MEMBER_FUNC { return TdotH; }
+    scalar_type getBdotH() NBL_CONST_MEMBER_FUNC { return BdotH; }
+
+    isocache_type iso_cache;
```

Comment:
these should be using `iso_cache.method()` not `iso_cache.member` because of the way we changed the concepts (they require getters be present, not members)